topics/admin/tutorials/wireguard-headscale/tutorial.md

```diff
@@ -0,0 +1,295 @@
+
+# Setting up the infrastructure
+
+{% include _includes/cyoa-choices.html option1="Tailscale" option2="Headscale" default="Tailscale" help="Here we provide a version of this tutorial that uses Headscale, if you want a more hands-on experience and feel like managing your own infrastructure." %}
+
+> ### {% icon hands_on %} Hands-on: Configuration files
+>
+> 1. Create a `ansible.cfg` file (next to your playbook) to [configure settings](https://docs.ansible.com/ansible/2.9/reference_appendices/config.html) like the inventory file (and save ourselves some typing!), or the Python interpreter to use:
+>
+>    {% raw %}
+>    ```diff
+>    --- /dev/null
+>    +++ b/ansible.cfg
+>    @@ -0,0 +1,6 @@
+>    +[defaults]
+>    +interpreter_python = /usr/bin/python3
+>    +inventory = hosts
+>    +retry_files_enabled = false
+>    +[ssh_connection]
+>    +pipelining = true
+>    {% endraw %}
+>    ```
+>    {: data-commit="Add ansible.cfg"}
+>
+> 2. Create the `hosts` inventory file if you have not done so yet.
+>
+>    > > ### {% icon code-in %} Input: Bash
+>    > > ```bash
+>    > > cat hosts
+>    > > ```
+>    > {: .code-in}
+>    >
+>    > > ### {% icon code-out %} Output: Bash
+>    > >
+>    > > Your hostname is probably different:
+>    > > <div class="Headscale" markdown="1">
+>    > > Pick one to be the `head`, and the rest to be the `tail`. The head will act as the coordination server, the `tail` will be the nodes that should talk to each other.
+>    > > {% raw %}
+>    > > ```diff
+>    > > --- /dev/null
+>    > > +++ b/hosts
+>    > > @@ -0,0 +1,5 @@
+>    > > +[head]
+>    > > +1-wg.galaxy.training
+>    > > +[tail]
+>    > > +2-wg.galaxy.training
+>    > > +3-wg.galaxy.training
+>    > > +4-wg.galaxy.training
+>    > > {% endraw %}
+>    > > ```
+>    > > {: data-commit="Add hosts"}
+>    > > </div>
+>    > >
+>    > > <div class="Tailscale" markdown="1">
+>    > > Place all of the nodes in a `tail` group
+>    > > ```diff
+>    > > --- /dev/null
+>    > > +++ b/hosts
+>    > > @@ -0,0 +1,5 @@
+>    > > +[tail]
+>    > > +1-wg.galaxy.training
+>    > > +2-wg.galaxy.training
+>    > > +3-wg.galaxy.training
+>    > > +4-wg.galaxy.training
+>    > > ```
+>    > > </div>
+>    > {: .code-out}
+>    {: .code-2col}
+{: .hands_on}
+
+<div class="Headscale" markdown="1">
+
+> ### {% icon details %} Consider using Tailscale
+> Tailscale has implemented the coordination server and infrastructure with much better, more robust infrastructure that you won't have to be responsible for. If you can get your university to pay for it, or grant money covering it, it's **probably worth it**. They add many new features often, and things like the mobile apps only work with their service.
+>
+> For a training event obviously we want something free and quick to setup and destroy, so, we're using [Headscale](https://github.com/juanfont/headscale) since it's free and we're just going to destroy it immediately, and no one will accidentally get billed ;)
```

Comment:
```suggestion
> For a training event obviously we want something free and quick to setup and destroy, so, we're using [Headscale](https://github.com/juanfont/headscale) since it's free and we're just going to destroy it immediately, and no one will accidentally get billed ;)
>
> Using Headscale will also teach you everything you need to know if you do choose to use Tailscale, which is simpler and has fewer components for you to manage yourself.
```